### PR TITLE
Angular 1.6 "Possibly unhandled rejection: canceled" fix. bug number #5890

### PR DESCRIPTION
**Pull request by dominusbelial**
using sebastian-zarzycki-es solution i have added ".catch(angular.noop)" to all the offending promises on the latest 4.0.2 uigrid for angular 1.6.1

### CODE REVIEW

**Comment by mportuga** [comment]
What is the bug number?

File: src/features/tree-base/js/tree-base.js [hunk 1]
```diff
@@ -1013,10 +1013,12 @@
         if ( grid.options.treeRowHeaderAlwaysVisible === false && grid.treeBase.numberLevels <= 0 ){
           newVisibility = false;
         }
-        if ( rowHeader.visible !== newVisibility ) {
-          rowHeader.visible = newVisibility;
-          rowHeader.colDef.visible = newVisibility;
-          grid.queueGridRefresh();
+        if ( rowHeader ) {
+          if ( rowHeader.visible !== newVisibility ) {
```
Comment (mportuga) [inline src/features/tree-base/js/tree-base.js]: why not do a single if statement?

```
if ( rowHeader  && rowHeader.visible !== newVisibility ) {
```

Comment (dominusbelial) [inline src/features/tree-base/js/tree-base.js]: bug #5890

Comment (dominusbelial) [inline src/features/tree-base/js/tree-base.js]: i will update to single if statement

**Comment by goleafs** [comment]
is this pull request going to be accepted?  would obviously prefer to stay on the main code base!

**Comment by mattbrunetti** [comment]
@goleafs Are you using this branch? If you've tested this let us know and it's more likely to be merged

**Comment by goleafs** [comment]
I made the same changes locally, have used it for a few days with no issues.

**Comment by mportuga** [comment]
@goleafs We will merge this. Unfortunately, we are currently busy with our day jobs right now, so we had to stop working on Ui-grid for a bit. We also are having a weird issue with the travis build. Once the travis build issue is resolved, we will see if we can get around to all of the open PRs. I am sorry for the inconvenience.

**Comment by Oceanswave** [comment]
+1

**Comment by ivorobkalo** [comment]
+1

**Comment by aamcatamney** [comment]
+1

**Comment by NageshTembhurnikar** [comment]
+1

**Pull request merged**